src/Orchestration/Adapter/DockerAPI.php

```diff
@@ -278,6 +278,75 @@ public function networkDisconnect(string $container, string $network, bool $forc
         return $result['code'] == 200;
     }
 
+    /**
+     * Get usage stats of containers
+     * 
+     * @param string $container
+     * @param array<string, string> $filters
+     * 
+     * @return array
+     */
+    public function getStats(string $container = null, array $filters = []): array
+    {
+        // List ahead of time, since API does not allow listing all usage stats
+        $containerIds = [];
+
+        if($container === null) {
+            $filtersSorted = [];
+
+            foreach($filters as $key => $value) {
+                $filtersSorted[$key] = [$value];
+            }
+    
+            $body = [
+                'all' => true,
+                'filters' => empty($filtersSorted) ? new stdClass() : json_encode($filtersSorted)
+            ];
+
+            $containers = \json_decode($this->call('http://localhost/containers/json?'.\http_build_query($body), 'GET')['response'], true);
+            $containerIds = \array_map(fn($c) => $c['Id'], $containers);
+        } else {
+            $containerIds[] = $container;
+        }
+
+        $list = [];
+
+        foreach($containerIds as $containerId) {
+            $result = $this->call('http://localhost/containers/' . $containerId . '/stats?stream=false', 'GET');
+
+            if ($result['code'] !== 200) {
+                throw new Orchestration($result['response']);
+            }
+
+            $stats = \json_decode($result['response'], true);
+
+            $cpuDelta = $stats['cpu_stats']['cpu_usage']['total_usage'] - $stats['precpu_stats']['cpu_usage']['total_usage'];
+            $systemCpuDelta = $stats['cpu_stats']['system_cpu_usage']  - $stats['precpu_stats']['system_cpu_usage'];
+
+            $networkIn = 0;
+            $networkOut = 0;
+            foreach ($stats['networks'] as $network) {
+                $networkIn += $network['rx_bytes'];
+                $networkOut += $network['tx_bytes'];
+            }
+
+            // Rx means Receive, and Tx means
+
+            $list[] = [
+                'id' => $stats['id'],
+                'name' => \ltrim($stats['name'], '/'), // Remove '/' prefix
+                'cpu' => 1, // TODO: Implement (I coudl not do it because Docker API does not give correct values)
+                // 'cpu' => ($cpuDelta / $systemCpuDelta) * $stats['cpu_stats']['online_cpus'] * 100.0,
```

Comment:
Can we clean this up a bit ?

Reply:
Removed comment. Formular should have worked, but didn't. So instead we stick with TODO.